### ISSUE TITLE
Single quotes break paren matching

### ISSUE THREAD

**Issue by shawwn** [Saigut/intellij-scheme#1]
Hello,

I use Vim emulation in IntelliJ.

One key I often use during Lisp development is %. This key jumps the cursor to the corresponding paren.

For example, `((foo) bar)` ... If the cursor is on the first paren and I press %, it jumps to the end of `bar`. If it's on the second paren, it jumps to the end of foo.

Your plugin for Scheme is nice, but single quotes break the above functionality. I'm not sure why.

for example,

```
('(foo) bar)
```

It seems like intelliJ thinks that single quote means "begin a new string", rather than "begin a quoted list."

I'm new to IntelliJ plugin development and would appreciate some advice on how to set up your project for debugging and testing. How do I build this myself? How do I publish changes?

I can learn, just point me in the right direction.

Thanks!

I notice you are Chinese and have not written much English. In case it's helpful, here is a Google Translated version:

你好，

我在IntelliJ中使用Vim仿真。

我经常在Lisp开发中使用的一个关键是`％`。 该键将光标跳转到相应的括号。

例如`（（foo）bar）`...如果光标位于第一个圆括号上，并按％，则跳转到“bar”的末尾。 如果它在第二个括号上，则跳转到foo的结尾。

Scheme的插件是好的，但单引号可以打破这一点。 我不知道为什么。

例如，

```
（'（foo）bar）
```

像intelliJ认为单引号意味着“开始一个新的字符串”，而不是“开始引用列表”。

我是新来IntelliJ插件开发，并希望有关如何设置您的项目进行调试和测试的建议：

- 如何在IntelliJ中构建您的项目？
- 如何发布更改？

我可以学习，只要指向正确的方向。

谢谢！

**Comment by Saigut** [comment]
Hello,

This plugin is modified from  [cmf/schemely](https://github.com/cmf/schemely) which code is somehow too old to compiled on newer IntelliJ platforms. 
I made the old code can be compiled on newer IntelliJ platforms, and modified some code of the Lexer and Parser, and also with other modifcations.
But this plugin is not well tested yet, and also have some issues that may be hard to be fixed for me. And I have little free time to work on this plugin recently.

I will test for the issue you met after work today.

If you try to build and debug this plugin, you can look at these urls for some help:
[https://www.jetbrains.org/intellij/sdk/docs/basics.html](https://www.jetbrains.org/intellij/sdk/docs/basics.html)
[https://www.jetbrains.org/intellij/sdk/docs/tutorials/custom_language_support_tutorial.html](https://www.jetbrains.org/intellij/sdk/docs/tutorials/custom_language_support_tutorial.html)
[https://www.jetbrains.org/intellij/sdk/docs/reference_guide/custom_language_support.html](https://www.jetbrains.org/intellij/sdk/docs/reference_guide/custom_language_support.html)

**Comment by shawwn** [comment]
Thank you for the help!

https://github.com/gregsh/Clojure-Kit might be useful for comparison, but
they use very different quoting rules. The ~ character is unquote, for
example.

**Comment by Saigut** [comment]
Thank you for reference!
And I am sorry to say that I was too late to work yesterday, so haven't test your issue yet. I'll try to test this issue in two days.

**Comment by shawwn** [comment]
That's ok. I hope your work is going well! I don't mind.

I usually work on IntelliJ plugins using gradlew, but this project has no
gradlew, so I don't know how to build it. I'll try to figure it out using
your references.

**Comment by Saigut** [comment]
Hello,

I tested the issue just now.

I have a question that, are you using plugin IdeaVim for Vim emulation?
I installed IdeaVim and met the same issue as you, but when I disabled the intellij scheme plugin, this issue still exist.

And then I disabled IdeaVim  and enabled scheme plugin, using `Ctrl + W` to match parentheses after `'` and before `'`, both match properly.

So it's seems that, the parentheses matching feature on hotkey % is implemented by IdeaVim rather than  this intellij scheme plugin.
